library/kotlin/io/envoyproxy/envoymobile/android/SharedPreferencesStore.kt

```diff
@@ -0,0 +1,24 @@
+package io.envoyproxy.envoymobile.android
+
+import android.content.SharedPreferences
+
+import io.envoyproxy.envoymobile.KeyValueStore
+
+class SharedPreferencesStore(sharedPreferences: SharedPreferences) : KeyValueStore {
```

Comment:
We don't today have a unit test setup that loads the Android platform APIs, because we don't today have a lot of Android-only or Android-specific code (by design). If/as we add more, we probably do want such a test suite, but this is mostly meant to be a trivial example implementation that guides others in using the API. That said, once we have a place for such tests to live, I agree we should cover this at the unit level for completeness' sake.

Reply:
I see, thank you for the context.

I would argue that we should have a test even for this trivial example since otherwise we cannot really tell whether it works or not the way we expect it to work - as in, I think that we are past the threshold when we can justify having an Android tests suite (if we have a need for Android code we have a need for Android tests). 

Could we create a GH issue for this if you do not want to do this as part of this PR? I would love us to start investing in testing more, otherwise it's going to become increasingly hard to continue to move forward without leaving broken stuff behind us.

Reply:
[Roboelectric](http://robolectric.org/) supports this today, and we use it in quite a few tests: https://github.com/envoyproxy/envoy-mobile/search?q=robolectric